src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -154,28 +175,28 @@ public int getFee() {
   /**
    * Generates Transaction XDR object.
    */
-  public org.stellar.sdk.xdr.Transaction toXdr() {
+  public TransactionV0 toXdr() {
```

Comment:
Isn't `xdr.Transaction` a default now?

```suggestion
  public Transaction toXdr() {
```

Reply:
until protocol 13 is enabled the sdks should still be generating v0 transactions. We will do another release after protocol 13 is enabled so that the sdks start generating v1 transactions. But, this method should be private the SDK users can get the xdr envelope. they don't need a public method to get the transaction from the envelope. I will change this method to private

Reply:
Our previous code using stellar api is broken now. I was told to use the latest version. One of place broken is that we were getting transaction from transaction envelope. Obviouly, it is broken now. Could you please let me know what is the alternative way to get the transaction from the envelope with current version? Are we suppose to keep compatability in version upgrading?

Reply:
@wumingmo can you please create an issue and provide some example code which demonstrates the broken behavior? thanks!

Reply:
Thanks, Tamirms, for quick reponse, I created a pull request before, https://github.com/stellar/java-stellar-sdk/issues/284, but was closed and told to use the latest version. Of course, after I use the 0.16.0, I got the compilation error. If possible, could you please let me know how to extract the transaction from the envelope with 0.16.0? Thanks